[client update] September 2021

zach2good opened the pull request:
<!-- remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm: -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

zach2good committed:
[client] Version update

zach2good committed:
[client] Titles

zach2good committed:
[client] Key Items (rough)

zach2good committed:
[client] Key Items (cleanup)

zach2good commented:
Looking for help with `npc_list.sql`, I've checked in the output from the shifting tool, so it needs a lot of work

zach2good commented:
I carefully resolved the merge conflicts, but I'm not sure if I've done `npc_list.sql - Mount Zhayolm` correctly, please pay close attention to that

zach2good commented:
`OVERDUE_BOOK_NOTIFICATION` needs checking (caught by CI)

claywar commented:
> `OVERDUE_BOOK_NOTIFICATION` needs checking (caught by CI)

There's a singular and a plural that exists, and these could be renamed regarding their quest.  Those were just recently converted to interaction

TeoTwawki commented:
> > `OVERDUE_BOOK_NOTIFICATION` needs checking (caught by CI)
> 
> There's a singular and a plural that exists, and these could be renamed regarding their quest. Those were just recently converted to interaction

IDs 10 and 149

claywar commented:
```
OVERDUE_BOOK_NOTIFICATIONS               = 10, -- Used in Early Bird Catches the Bookworm (Art for Everyone)
OVERDUE_BOOK_NOTIFICATION                = 149, -- Used in Chasing Tales (A Song of Love)
```

zach2good committed:
[client] Fix KI name conflict

TeoTwawki committed:
NPC ID updates part 1 for client version 30210904_2

Note:
- Starter city zones need NPCs fully recapped.
- Starter zone mobs need fully recapped.

SE removed so many event objects and added so many new ones that there is no feasible way to automate this nor can these be hand edited in a way that makes any sense without fresh captures of the same version.

TeoTwawki commented:
I updated NPC list as much as I could. The starter zones have both mob and NPC changes that are still not done, we will need caps for those.

I also did not fix the IDs files where there is an NPC ID used. Those need to match the NPC list. Pretty sure Attohwa Chasm's gas walls are busted from several updates ago and it just hasn't been reported on because people aren't updated.

Advise against committing the "fix me"s in the future. They were mostly incorrect anyway - felt like it managed to only flag ones where the ID hadn't actually changed.

TeoTwawki committed:
Spot fix mistakes I made in npc list update

zach2good commented:
I can squash mine and your commits together to make sure the FIXMEs don't make it into the history

TeoTwawki commented:
> I can squash mine and your commits together to make sure the FIXMEs don't make it into the history

It being in the history after the fact won't pose a problem, bit of a pain while syncing my wip with your wip tho :)

claywar committed:
Add additional NPC ID shifts, fix IDs.lua for Beaucedine and Palborough

claywar committed:
Merge pull request #743 from claywar/sep-update

Add additional NPC ID shifts, fix IDs.lua for Beaucedine and Palborough

TeoTwawki commented on scripts/globals/keyitems.lua:
```diff
@@ -289,7 +288,7 @@ xi.keyItem =
     MANUSTERY_RING                           = 281,
     GLOVE_OF_PERPETUAL_TWILIGHT              = 282,
-    ANCIENT_SANDORIAN_TABLET                 = 283,
+    ANCIENT_SAN_DORIAN_TABLET                = 283,
```
~~we strip out `'` not replace with underscore, so this needs to return to what it was~~

ugh nvm not where the `'` is at and I am simply wrong

TeoTwawki commented on scripts/globals/keyitems.lua:
```diff
@@ -1716,7 +1718,7 @@ xi.keyItem =
     THRONE_ROOM_DAWNDROP                     = 1727,
     WALK_OF_ECHOES_DAWNDROP                  = 1728,
-    SANDORIA_DAWNDROP                        = 1729,
+    SAN_DORIA_DAWNDROP                       = 1729,
```
same here and all similar instances
